### DISCUSSION TITLE
Fail to install on chromebook

### DISCUSSION THREAD

**Discussion by xzhang222** [pymupdf/PyMuPDF#1135] [Installation Topics]
Try to install PyMuPDF on chromebook in linux container by "sudo pip3 install PyMuPDF", get following error messages:

Failed building wheel for PyMuPDF
  Running setup.py clean for PyMuPDF
Failed to build PyMuPDF
Installing collected packages: PyMuPDF
  Running setup.py install for PyMuPDF ... error
    Complete output from command /usr/bin/python3 -u -c "import setuptools, tokenize;__file__='/tmp/pip-install-1chhb5fg/PyMuPDF/setup.py';f=getattr(tokenize, 'open', open)(__file__);code=f.read().replace('\r\n', '\n');f.close();exec(compile(code, __file__, 'exec'))" install --record /tmp/pip-record-fig12_46/install-record.txt --single-version-externally-managed --compile:
    running install
    running build
    running build_py
    creating build
    creating build/lib.linux-x86_64-3.7
    creating build/lib.linux-x86_64-3.7/fitz
    copying fitz/__init__.py -> build/lib.linux-x86_64-3.7/fitz
    copying fitz/fitz.py -> build/lib.linux-x86_64-3.7/fitz
    copying fitz/utils.py -> build/lib.linux-x86_64-3.7/fitz
    copying fitz/__main__.py -> build/lib.linux-x86_64-3.7/fitz
    running build_ext
    building 'fitz._fitz' extension
    creating build/temp.linux-x86_64-3.7
    creating build/temp.linux-x86_64-3.7/fitz
    x86_64-linux-gnu-gcc -pthread -DNDEBUG -g -fwrapv -O2 -Wall -g -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -fPIC -I/usr/include/mupdf -I/usr/local/include/mupdf -Imupdf/thirdparty/freetype/include -I/usr/include/freetype2 -I/usr/include/python3.7m -c fitz/fitz_wrap.c -o build/temp.linux-x86_64-3.7/fitz/fitz_wrap.o
    fitz/fitz_wrap.c:2754:10: fatal error: fitz.h: No such file or directory
     #include <fitz.h>
              ^~~~~~~~
    compilation terminated.
    error: command 'x86_64-linux-gnu-gcc' failed with exit status 1

If there ways to fix this problem?

**Comment by JorjMcKie** [comment]
Bad timing! I am just uploading v1.18.15, and so far only the source is done.
Either wait some time or install pymupdf==1.18.14

Von meinem iPhone gesendet

Am 10/7/21 um 10:22 schrieb xzhang222 ***@***.***>:

﻿

Try to install PyMuPDF on chromebook in linux container by "sudo pip3 install PyMuPDF", get following error messages:

Failed building wheel for PyMuPDF
Running setup.py clean for PyMuPDF
Failed to build PyMuPDF
Installing collected packages: PyMuPDF
Running setup.py install for PyMuPDF ... error
Complete output from command /usr/bin/python3 -u -c "import setuptools, tokenize;file='/tmp/pip-install-1chhb5fg/PyMuPDF/setup.py';f=getattr(tokenize, 'open', open)(file);code=f.read().replace('\r\n', '\n');f.close();exec(compile(code, file, 'exec'))" install --record /tmp/pip-record-fig12_46/install-record.txt --single-version-externally-managed --compile:
running install
running build
running build_py
creating build
creating build/lib.linux-x86_64-3.7
creating build/lib.linux-x86_64-3.7/fitz
copying fitz/init.py -> build/lib.linux-x86_64-3.7/fitz
copying fitz/fitz.py -> build/lib.linux-x86_64-3.7/fitz
copying fitz/utils.py -> build/lib.linux-x86_64-3.7/fitz
copying fitz/main.py -> build/lib.linux-x86_64-3.7/fitz
running build_ext
building 'fitz._fitz' extension
creating build/temp.linux-x86_64-3.7
creating build/temp.linux-x86_64-3.7/fitz
x86_64-linux-gnu-gcc -pthread -DNDEBUG -g -fwrapv -O2 -Wall -g -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -fPIC -I/usr/include/mupdf -I/usr/local/include/mupdf -Imupdf/thirdparty/freetype/include -I/usr/include/freetype2 -I/usr/include/python3.7m -c fitz/fitz_wrap.c -o build/temp.linux-x86_64-3.7/fitz/fitz_wrap.o
fitz/fitz_wrap.c:2754:10: fatal error: fitz.h: No such file or directory
#include <fitz.h>
^~~~~~~~
compilation terminated.
error: command 'x86_64-linux-gnu-gcc' failed with exit status 1

If there ways to fix this problem?

**Reply by xzhang222** [reply]
Thanks for quick response. I tried 1.18.14 and get same problem as shown below. I feel this problem is specific to linux on chromebook, as I find installation on Macbook goes correctly.

xdzhang2@penguin:~$ sudo pip3 install pymupdf==1.18.14
Collecting pymupdf==1.18.14
  Using cached https://files.pythonhosted.org/packages/41/f6/dbefe3d6949fa81fb7bcac9141e4345330d272724718ac5a6af78297498b/PyMuPDF-1.18.14.tar.gz
Building wheels for collected packages: pymupdf
  Running setup.py bdist_wheel for pymupdf ... error
  Complete output from command /usr/bin/python3 -u -c "import setuptools, tokenize;__file__='/tmp/pip-install-y_zuj2x6/pymupdf/setup.py';f=getattr(tokenize, 'open', open)(__file__);code=f.read().replace('\r\n', '\n');f.close();exec(compile(code, __file__, 'exec'))" bdist_wheel -d /tmp/pip-wheel-ioscfyvg --python-tag cp37:
  running bdist_wheel
  running build
  running build_py
  creating build
  creating build/lib.linux-x86_64-3.7
  creating build/lib.linux-x86_64-3.7/fitz
  copying fitz/__init__.py -> build/lib.linux-x86_64-3.7/fitz
  copying fitz/fitz.py -> build/lib.linux-x86_64-3.7/fitz
  copying fitz/utils.py -> build/lib.linux-x86_64-3.7/fitz
  copying fitz/__main__.py -> build/lib.linux-x86_64-3.7/fitz
  running build_ext
  building 'fitz._fitz' extension
  creating build/temp.linux-x86_64-3.7
  creating build/temp.linux-x86_64-3.7/fitz
  x86_64-linux-gnu-gcc -pthread -DNDEBUG -g -fwrapv -O2 -Wall -g -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -fPIC -I/usr/include/mupdf -I/usr/local/include/mupdf -Imupdf/thirdparty/freetype/include -I/usr/include/freetype2 -I/usr/include/python3.7m -c fitz/fitz_wrap.c -o build/temp.linux-x86_64-3.7/fitz/fitz_wrap.o
  fitz/fitz_wrap.c:2754:10: fatal error: fitz.h: No such file or directory
   #include <fitz.h>
            ^~~~~~~~
  compilation terminated.
  error: command 'x86_64-linux-gnu-gcc' failed with exit status 1
  
  ----------------------------------------
  Failed building wheel for pymupdf
  Running setup.py clean for pymupdf
Failed to build pymupdf
Installing collected packages: pymupdf
  Running setup.py install for pymupdf ... error
    Complete output from command /usr/bin/python3 -u -c "import setuptools, tokenize;__file__='/tmp/pip-install-y_zuj2x6/pymupdf/setup.py';f=getattr(tokenize, 'open', open)(__file__);code=f.read().replace('\r\n', '\n');f.close();exec(compile(code, __file__, 'exec'))" install --record /tmp/pip-record-abgutj43/install-record.txt --single-version-externally-managed --compile:
    running install
    running build
    running build_py
    creating build
    creating build/lib.linux-x86_64-3.7
    creating build/lib.linux-x86_64-3.7/fitz
    copying fitz/__init__.py -> build/lib.linux-x86_64-3.7/fitz
    copying fitz/fitz.py -> build/lib.linux-x86_64-3.7/fitz
    copying fitz/utils.py -> build/lib.linux-x86_64-3.7/fitz
    copying fitz/__main__.py -> build/lib.linux-x86_64-3.7/fitz
    running build_ext
    building 'fitz._fitz' extension
    creating build/temp.linux-x86_64-3.7
    creating build/temp.linux-x86_64-3.7/fitz
    x86_64-linux-gnu-gcc -pthread -DNDEBUG -g -fwrapv -O2 -Wall -g -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -fPIC -I/usr/include/mupdf -I/usr/local/include/mupdf -Imupdf/thirdparty/freetype/include -I/usr/include/freetype2 -I/usr/include/python3.7m -c fitz/fitz_wrap.c -o build/temp.linux-x86_64-3.7/fitz/fitz_wrap.o
    fitz/fitz_wrap.c:2754:10: fatal error: fitz.h: No such file or directory
     #include <fitz.h>
              ^~~~~~~~
    compilation terminated.
    error: command 'x86_64-linux-gnu-gcc' failed with exit status 1

**Comment by JorjMcKie** [comment]
Ah ok. Then obviously this Linux is not supported by any of the wheels, so that you have to install from sources - see docu. *Or* you don’t have the most current PIP installed. One of the very problems.

Von meinem iPhone gesendet

Am 10/7/21 um 10:43 schrieb xzhang222 ***@***.***>:

﻿

Thanks for quick response. I tried 1.18.14 and get same problem as shown below. I feel this problem is specific to linux on chromebook, as I find installation on Macbook goes correctly.

***@***.***:~$ sudo pip3 install pymupdf==1.18.14
Collecting pymupdf==1.18.14
Using cached https://files.pythonhosted.org/packages/41/f6/dbefe3d6949fa81fb7bcac9141e4345330d272724718ac5a6af78297498b/PyMuPDF-1.18.14.tar.gz
Building wheels for collected packages: pymupdf
Running setup.py bdist_wheel for pymupdf ... error
Complete output from command /usr/bin/python3 -u -c "import setuptools, tokenize;file='/tmp/pip-install-y_zuj2x6/pymupdf/setup.py';f=getattr(tokenize, 'open', open)(file);code=f.read().replace('\r\n', '\n');f.close();exec(compile(code, file, 'exec'))" bdist_wheel -d /tmp/pip-wheel-ioscfyvg --python-tag cp37:
running bdist_wheel
running build
running build_py
creating build
creating build/lib.linux-x86_64-3.7
creating build/lib.linux-x86_64-3.7/fitz
copying fitz/init.py -> build/lib.linux-x86_64-3.7/fitz
copying fitz/fitz.py -> build/lib.linux-x86_64-3.7/fitz
copying fitz/utils.py -> build/lib.linux-x86_64-3.7/fitz
copying fitz/main.py -> build/lib.linux-x86_64-3.7/fitz
running build_ext
building 'fitz._fitz' extension
creating build/temp.linux-x86_64-3.7
creating build/temp.linux-x86_64-3.7/fitz
x86_64-linux-gnu-gcc -pthread -DNDEBUG -g -fwrapv -O2 -Wall -g -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -fPIC -I/usr/include/mupdf -I/usr/local/include/mupdf -Imupdf/thirdparty/freetype/include -I/usr/include/freetype2 -I/usr/include/python3.7m -c fitz/fitz_wrap.c -o build/temp.linux-x86_64-3.7/fitz/fitz_wrap.o
fitz/fitz_wrap.c:2754:10: fatal error: fitz.h: No such file or directory
#include <fitz.h>
^~~~~~~~
compilation terminated.
error: command 'x86_64-linux-gnu-gcc' failed with exit status 1

________________________________

Failed building wheel for pymupdf
Running setup.py clean for pymupdf
Failed to build pymupdf
Installing collected packages: pymupdf
Running setup.py install for pymupdf ... error
Complete output from command /usr/bin/python3 -u -c "import setuptools, tokenize;file='/tmp/pip-install-y_zuj2x6/pymupdf/setup.py';f=getattr(tokenize, 'open', open)(file);code=f.read().replace('\r\n', '\n');f.close();exec(compile(code, file, 'exec'))" install --record /tmp/pip-record-abgutj43/install-record.txt --single-version-externally-managed --compile:
running install
running build
running build_py
creating build
creating build/lib.linux-x86_64-3.7
creating build/lib.linux-x86_64-3.7/fitz
copying fitz/init.py -> build/lib.linux-x86_64-3.7/fitz
copying fitz/fitz.py -> build/lib.linux-x86_64-3.7/fitz
copying fitz/utils.py -> build/lib.linux-x86_64-3.7/fitz
copying fitz/main.py -> build/lib.linux-x86_64-3.7/fitz
running build_ext
building 'fitz._fitz' extension
creating build/temp.linux-x86_64-3.7
creating build/temp.linux-x86_64-3.7/fitz
x86_64-linux-gnu-gcc -pthread -DNDEBUG -g -fwrapv -O2 -Wall -g -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -fPIC -I/usr/include/mupdf -I/usr/local/include/mupdf -Imupdf/thirdparty/freetype/include -I/usr/include/freetype2 -I/usr/include/python3.7m -c fitz/fitz_wrap.c -o build/temp.linux-x86_64-3.7/fitz/fitz_wrap.o
fitz/fitz_wrap.c:2754:10: fatal error: fitz.h: No such file or directory
#include <fitz.h>
^~~~~~~~
compilation terminated.
error: command 'x86_64-linux-gnu-gcc' failed with exit status 1
